pkg/kube/client.go

```diff
@@ -674,48 +944,95 @@ func createPatch(target *resource.Info, current runtime.Object, threeWayMergeFor
 	return patch, types.StrategicMergePatchType, err
 }
 
-func updateResource(_ *Client, target *resource.Info, currentObj runtime.Object, force, threeWayMergeForUnstructured bool) error {
-	var (
-		obj    runtime.Object
-		helper = resource.NewHelper(target.Client, target.Mapping).WithFieldManager(getManagedFieldsManager())
-		kind   = target.Mapping.GroupVersionKind.Kind
-	)
+func replaceResource(target *resource.Info, fieldValidationDirective FieldValidationDirective) error {
 
-	// if --force is applied, attempt to replace the existing resource with the new object.
-	if force {
-		var err error
-		obj, err = helper.Replace(target.Namespace, target.Name, true, target.Object)
-		if err != nil {
-			return fmt.Errorf("failed to replace object: %w", err)
-		}
-		slog.Debug("replace succeeded", "name", target.Name, "initialKind", currentObj.GetObjectKind().GroupVersionKind().Kind, "kind", kind)
-	} else {
-		patch, patchType, err := createPatch(target, currentObj, threeWayMergeForUnstructured)
-		if err != nil {
-			return fmt.Errorf("failed to create patch: %w", err)
-		}
+	helper := resource.NewHelper(target.Client, target.Mapping).
+		WithFieldValidation(string(fieldValidationDirective)).
+		WithFieldManager(getManagedFieldsManager())
 
-		if patch == nil || string(patch) == "{}" {
-			slog.Debug("no changes detected", "kind", kind, "name", target.Name)
-			// This needs to happen to make sure that Helm has the latest info from the API
-			// Otherwise there will be no labels and other functions that use labels will panic
-			if err := target.Get(); err != nil {
-				return fmt.Errorf("failed to refresh resource information: %w", err)
-			}
-			return nil
-		}
-		// send patch to server
-		slog.Debug("patching resource", "kind", kind, "name", target.Name, "namespace", target.Namespace)
-		obj, err = helper.Patch(target.Namespace, target.Name, patchType, patch, nil)
-		if err != nil {
-			return fmt.Errorf("cannot patch %q with kind %s: %w", target.Name, kind, err)
+	obj, err := helper.Replace(target.Namespace, target.Name, true, target.Object)
+	if err != nil {
+		return fmt.Errorf("failed to replace object: %w", err)
+	}
+
+	if err := target.Refresh(obj, true); err != nil {
+		return fmt.Errorf("failed to refresh object after replace: %w", err)
+	}
+
+	return nil
+
+}
+
+func patchResourceClientSide(target *resource.Info, original runtime.Object, threeWayMergeForUnstructured bool) error {
```

Comment:
this was split fro the previous `updateResource` function: https://github.com/helm/helm/pull/31030/files#diff-137bba79988de442c1fde0fc29520d6d7e5a5239ceac82f5439a5bf513ebb2d4L693